core/tee/tee_rpmb_fs.c

```diff
@@ -437,12 +437,14 @@ static void tee_rpmb_free(struct tee_rpmb_mem *mem)
 	if (!mem)
 		return;
 
-	thread_rpc_free_payload(mem->phreq_cookie);
-	thread_rpc_free_payload(mem->phresp_cookie);
-	mem->phreq = 0;
-	mem->phreq_cookie = 0;
-	mem->phresp = 0;
-	mem->phresp_cookie = 0;
+	if (mem->phreq) {
```

Comment:
Physical address 0 could be valid. But we know on the other hand that a cookie with value 0 can be valid as the driver is using a valid virtual address as cookie. This isn't documented in the interface though.